R/build_html_mans.R

```diff
@@ -0,0 +1,187 @@
+        db <- lapply(files, readRDS)
+        names(db) <- packages
+    } else {
+        db <- readRDS(aliases_db_file)
+        ## Drop entries in db not in package web area.
+        db <- db[!is.na(match(names(db), packages))]
+        ## Update entries for which aliases file is more recent than the
+        ## db file.
+        mtimes <- file.mtime(files)
+        files <- files[mtimes >= file.mtime(aliases_db_file)]
+        db[basename(dirname(files))] <- lapply(files, readRDS)
+    }
+
+    db[sort(names(db))]
+}
+
+is_file <- function(x) file.exists(x) && !file.info(x)[["isdir"]]
+
+#' @rdname html-manuals
+#'
+#' @param rdxrefs_db_file `character(1)` The file path to `rdxrefs.rds` file
+#'   generated by the `build_db_from_source` function.
+#'
+#' @examples
+#' if (interactive()) {
+#'     src_destDir <- "~/minibioc/packages/3.20/bioc/"
+#'     web_dir <- file.path(src_destDir, "web", "packages")
+#'
+#'     meta_folder <- file.path(contrib.url(src_destDir), "Meta")
+#'     if (!dir.exists(meta_folder)) dir.create(meta_folder, recursive = TRUE)
+#'     rdxrefs_db_file <- file.path(meta_folder, "rdxrefs.rds")
+#'
+#'     meta_rdxrefs_db <- build_meta_rdxrefs_db(web_dir, rdxrefs_db_file)
+#'
+#'     saveRDS(meta_rdxrefs_db, rdxrefs_db_file, version = 2)
+#' }
+#' @export
+build_meta_rdxrefs_db <-
+    function(web_dir, rdxrefs_db_file, force = FALSE)
+{
+    files <- Sys.glob(file.path(web_dir, "*", "rdxrefs.rds"))
+    packages <- basename(dirname(files))
+    if(force || !is_file(rdxrefs_db_file)) {
+        db <- lapply(files, readRDS)
+        names(db) <- packages
+    } else {
+        db <- readRDS(rdxrefs_db_file)
+        ## Drop entries in db not in package web area.
+        db <- db[!is.na(match(names(db), packages))]
+        ## Update entries for which rdxrefs file is more recent than the
+        ## db file.
+        mtimes <- file.mtime(files)
+        files <- files[mtimes >= file.mtime(rdxrefs_db_file)]
+        db[basename(dirname(files))] <- lapply(files, readRDS)
+    }
+
+    db[sort(names(db))]
+}
+
+#' @rdname html-manuals
+#'
+#' @param package_dirs `character()` A vector of local package directories.
+#'
+#' @examples
+#' if (interative()) {
+#'     library(BiocPkgTools)
+#'     bioc_sub <- pkgBiocDeps(
+#'         "SummarizedExperiment", pkgType = "software",
+#'         recursive = TRUE, only.bioc = TRUE
+#'     )
+#'     bioc_sub <- unlist(bioc_sub, use.names = FALSE)
+#'
+#'     ## generate from Bioc package source dirs
+#'     packages <- file.path(normalizePath("~/bioc"), bioc_sub)
+#'     src_destDir <- "~/minibioc/packages/3.20/bioc"
+#'
+#'     build_html_mans(packages, src_destDir)
+#' }
+#' @export
+build_html_mans <- function(package_dirs, src_destDir) {
+    packages <- basename(package_dirs)
+    man_dir <- file.path(src_destDir, "manuals")
+    if (!dir.exists(man_dir)) dir.create(man_dir, recursive = TRUE)
+    outfiles <- file.path(man_dir, packages, "man", paste0(packages, ".html"))
```

Comment:
I think makes `manuals/aPkg/man/aPkg.html`. I'm not sure if this is better and maybe my comment was confusing. Before we make further changes, let's discuss with Lori how they can be made available on the website.